kernel/src/hil/accel.rs

```diff
@@ -0,0 +1,52 @@
+//! Interface for Accelerators
+
+use crate::common::leasable_buffer::LeasableBuffer;
+use crate::ErrorCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a, const T: usize> {
+    /// This callback is called when the binary data has been loaded
+    /// On error or success `input` will contain a reference to the original
+    /// data supplied to `load_binary()`.
+    fn binary_load_done(&'a self, result: Result<(), ErrorCode>, input: &'static mut [u8]);
+
+    /// This callback is called when a operation is computed.
+    /// On error or success `output` will contain a reference to the original
+    /// data supplied to `run()`.
+    fn op_done(&'a self, result: Result<(), ErrorCode>, output: &'static mut [u8; T]);
+}
+
+/// A generic accellerator. This can be used to accelerate any type of
+/// operation.
+pub trait Accel<'a, const T: usize> {
+    /// Set the client instance which will receive
+    fn set_client(&'a self, client: &'a dyn Client<'a, T>);
+
+    /// Load the acceleration binary data into the accelerator.
+    /// This data will be accelerator specific and could be an
+    /// elf file which will be run or could be binary settings used to
+    /// configure the accelerator.
+    /// This function can be called multiple times if multiple binary blobs
+    /// are required.
+    /// There is no guarantee the data has been written until the `binary_load_done()`
+    /// callback is fired.
+    /// On error the return value will contain a return code and the original data
+    fn load_binary(
+        &self,
+        input: LeasableBuffer<'static, u8>,
+    ) -> Result<(), (ErrorCode, &'static mut [u8])>;
```

Comment:
It seems strange to pass in a `LeasableBuffer` but get back a slice in the error case

Reply:
This is the same thing we do with the HMAC. The idea is to pass in a buffer and a length and then we just return the buffer. In this case though we skip the length argument by using the `LeasableBuffer`